### PR TITLE
[12.0] [IMP] Banking Mandate: set reference at creation if falsy value in dict

### LINKED ISSUE #592: [12.0] [RFC] Banking Mandate: unique reference at validation
It's currently possible to import a mandate with no reference and validate it.

What about making the `unique_mandate_reference` readonly (in Python code, not only in views) and generating the reference at validation instead of creation (closer than what is done on invoices) + a constraint making the reference required in other states than draft.

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
fixes https://github.com/OCA/bank-payment/issues/592

### CODE REVIEW

File: account_banking_mandate/models/account_banking_mandate.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def _check_valid_state(self):
 
     @api.model
     def create(self, vals=None):
-        if vals.get('unique_mandate_reference', 'New') == 'New':
+        if not vals.get('unique_mandate_reference'):
```
Comment (sbidoul) [inline account_banking_mandate/models/account_banking_mandate.py]: @ThomasBinsfeld if the user provides `New` I think you must preserve the current behaviour.

Comment (ThomasBinsfeld) [reply account_banking_mandate/models/account_banking_mandate.py]: Done

**Review by guewen — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Can you add a test for avoiding regressions?

**Comment by ThomasBinsfeld** [comment]
@pedrobaeza Done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**